### ISSUE TITLE
Test accuracy

### ISSUE THREAD

**Issue by whjxnyzh** [facebook/C3D#9]
```
E0529 11:24:15.056433 13957 test_net.cpp:51] Batch 1373, accuracy: 0.8
E0529 11:24:16.577850 13957 test_net.cpp:51] Batch 1374, accuracy: 0.833333
E0529 11:24:18.104264 13957 test_net.cpp:51] Batch 1375, accuracy: 0.833333
E0529 11:24:19.628096 13957 test_net.cpp:51] Batch 1376, accuracy: 0.833333
E0529 11:24:21.147753 13957 test_net.cpp:51] Batch 1377, accuracy: 0.833333
E0529 11:24:22.675658 13957 test_net.cpp:51] Batch 1378, accuracy: 0.733333
E0529 11:24:24.198937 13957 test_net.cpp:51] Batch 1379, accuracy: 0.8
E0529 11:24:25.720679 13957 test_net.cpp:51] Batch 1380, accuracy: 0.8
E0529 11:24:27.248215 13957 test_net.cpp:51] Batch 1381, accuracy: 0.866667
E0529 11:24:28.769614 13957 test_net.cpp:51] Batch 1382, accuracy: 0.833333
E0529 11:24:30.297591 13957 test_net.cpp:51] Batch 1383, accuracy: 0.766667
E0529 11:24:30.297626 13957 test_net.cpp:54] Test accuracy: 0.786777
```
I run ucf101_testing.sh
I change batch size from 50 to 30,
so I have 1384 batches in total
Test accuracy is 0.786777, is this ok?

**Comment by dutran** [comment]
@whjxnyzh this looks OK to me. Note that this is clip-based accuracy. If you want to make video-based accuracy. You can use feature extraction tool, to extract soft-max predictions (e.g. layer "prob" activations) and then aggregate these soft-max predictions to make video-based predictions (and so accuracy). A simple way to aggregate is to average the clip-based predictions to make video-based predictions.

**Comment by whjxnyzh** [comment]
@dutran  Thank you very much, i will try it out.

**Comment by whjxnyzh** [comment]
@dutran , Great, "A simple way to aggregate is to average the clip-based predictions to make video-based predictions."
Now,the accuray is 0.8242135871
```
video: v_WritingOnBoard_g04_c01 predicted label: 99 True label: 99
video: v_TrampolineJumping_g02_c05 predicted label: 93 True label: 93
video: v_CricketBowling_g02_c06 predicted label: 22 True label: 22
video: v_BoxingPunchingBag_g02_c04 predicted label: 17 True label: 16
video: v_HandstandWalking_g07_c03 predicted label: 51 True label: 37
video: v_CliffDiving_g07_c06 predicted label: 82 True label: 21
video: v_Skiing_g05_c01 predicted label: 80 True label: 80
video: v_YoYo_g03_c05 predicted label: 55 True label: 100
video: v_PlayingPiano_g05_c03 predicted label: 63 True label: 63
video: v_PlayingPiano_g07_c01 predicted label: 63 True label: 63
video: v_Swing_g03_c03 predicted label: 88 True label: 88
video level accuray: 0.8242135871
```
Thank you very much, this is my first experiment about video using deep learning method

**Comment by mzolfaghari** [comment]
@whjxnyzh @dutran How did you get these results?? When I run train_ucf101.sh it gives me around 45% accuracy!! But you get around 80% accuracy!! 
@dutran also in the C3DUserGuide said that we get around 45% accuracy by running train_ucf101.sh!! 
I'm confused about the accuracies!! In the paper we have 82% accuracy and here @whjxnyzh get's 82% accuracy but other guys (e.g. https://github.com/facebook/C3D/issues/42 ), me and C3DUserGuide reported ~45% accuracy!! 
I really appreciate if you clarify this.

Best,
- 
Mohammadreza

**Comment by dutran** [comment]
@mzolfaghari If you train from scratch, it is supposed to be around 45%. If you fine-tune, you can get higher accuracy 77-78% on clip-accuracy and 82-83% on video-accuracy

**Comment by holyseven** [comment]
Hi again @dutran ,  I am confused again.... ;( I am reading your article _Learning Spatiotemporal Features with 3D Convolutional Networks_. In Table 2, C3D(fine-tuned from I380K pre-trained model), the accuracy for one clip hit@1 is 46.1%, not much higher than clip hit@1 of C3D (trained from scratch). Is there something I missed in your article or I misunderstood? Thank you in advance for your help.

**Comment by dutran** [comment]
There is nothing you missed. As I380K and Sports1M, fine-tuning can help but not making a big difference.

**Comment by holyseven** [comment]
But you said "If you fine-tune, you can get higher accuracy 77-78% on clip-accuracy and 82-83% on video-accuracy".

**Comment by dutran** [comment]
Yes, you fine-tune, then use fine-tuned model to extract prob, then make predictions. Does that make sense to you?

**Comment by holyseven** [comment]
So this is what I think. Training from scratch on UCF101, then output directly from convnet, the clip-accuracy is around 44.9%. Fine-tuning on UCF101 from the pretrained model on I380K, the clip-accuracy is around 46.1%. But using fine-tuned model, with the extracted features and SVM, we could get 77-78% on clip-accuracy. Is this correct?

**Comment by dutran** [comment]
No, 44.9 and 46.1 is testing on Sports1M, not UCF101.

But if you training from scratch from UCF101, you will get ~45% (clip accuracy), if you fine-tune the model (which was trained on Sports1M), you will get ~ 77-78% (clip accuracy).

In any case, you can use the model either provided by us (which is pre-trained on Sports1M) or the your own fine-tuned model (on your own data/task) to extract fc6 or prob to make prediction. prob is directly related to the task of the trained net where fc6 can be further train which in our paper using a linear SVM. Hope that helps.

**Comment by holyseven** [comment]
Thank you very much

**Comment by mariankyoussef** [comment]
"A simple way to aggregate is to average the clip-based predictions to make video-based predictions."
Excuse me, can I know how to take the output of averaging clip accuracy? I mean, if 3 out of 5 clips got right predictions and the other 2 were wrong, this gives an average of 60% per video, what's next?

I mean, if I took 60% for this video, and say, 75% for another video, and then take the average over all videos, this will give the same output as clip-level.. Should I take a threshold?
